collective_coverage.py

```diff
@@ -0,0 +1,89 @@
+import inspect
+import getopt
+import sys
+import os
+
+
+def get_full_path(filename):
+    current_file_path = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+    return os.path.join(current_file_path, filename)
+
+
+def process_arguments(argv):
+
+    try:
+        opts, args = getopt.getopt(argv,"hi:o:f:c:",[])
+
+    except getopt.GetoptError:
+        sys.exit('collective_coverage.py -i <input_dir_with_test_cases> -o <output_dir> -f <program_object_file_path> -c <collective_coverage_file> ')
+
+    for opt, arg in opts:
+        if opt == '-h':
+            print('collective_coverage.py -i <input_dir_with_test_cases> -o <output_dir> -f <program_object_file_path> -c <collective_coverage_file>')
+            sys.exit()
+        elif opt == '-i':
+            input_dir = arg
+        elif opt == '-o':
+            output_dir = arg
+        elif opt == '-f':
+            object_file = arg
+        elif opt == '-c':
+            collective_coverage_path = arg
+            
+    return input_dir, output_dir, object_file, collective_coverage_path
+
+
+def run_program_on_tests(input_dir, output_dir, object_file):
+
+    bb_coverage_path = get_full_path('./counter_BBid_coverage')
+    test_inputs = os.listdir(input_dir)
+    object_file_name = object_file.strip().split('/')[-1].split('.')[0]
+
+    for input_name in test_inputs:
+        input_file = os.path.join(input_dir, input_name)
+        input_name = input_name.strip().split('.')[0]
+        if not os.path.isfile(input_file):
+            continue
```

Comment:
Let's log an error here since this shouldn't be the normal case.

Reply:
Should we just log that something is wrong or should we exit the run?

Reply:
I think existing makes sense unless you've observed some behavior that makes this a more common case.